src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -492,7 +531,22 @@ public T WithAuthority(Uri authorityUri, bool validateAuthority = true)
             string tenant,
             bool validateAuthority = true)
         {
-            Config.AuthorityInfo = AuthorityInfo.FromAadAuthority(azureCloudInstance, tenant, validateAuthority);
+            if (string.IsNullOrWhiteSpace(tenant))
+            {
+                throw new ArgumentNullException(nameof(tenant));
+            }
+
+            if (!string.IsNullOrWhiteSpace(Config.TenantId))
+            {
+                throw new MsalClientException(
+                    MsalError.DuplicateTenantIdConfig,
+                    MsalErrorMessage.DuplicateTenantIdConfig(Config.TenantId, tenant));
+            }
+
+            Config.AzureCloudInstance = azureCloudInstance;
```

Comment:
The fix for this bug is to delay computing the AuthorityInfo in case the developer uses the enums. This is consistent to the developer using the Options object.

Reply:
Delay? to the Build.() method?

Reply:
Yes, postpone creation of `AuthorityInfo` and validation to the Build() method, to be able to take into account `WithTenantId`